src/page-modules/departures/server/journey-planner/index.ts

```diff
@@ -358,3 +383,54 @@ type RecursivePartial<T> = {
     ? RecursivePartial<T[P]>
     : T[P];
 };
+
+const mapToMapLegs = (
+  pointsOnLink: GraphQlPointsOnLink | undefined,
+  transportMode: TransportModeType,
+  transportSubmode: TransportSubmodeType | undefined,
+  fromStopPlace?: { id: string; latitude?: number; longitude?: number },
+) => {
+  if (!pointsOnLink || !pointsOnLink.points) return [];
+  const points = polyline.decode(pointsOnLink.points);
+  const fromCoordinates: [number, number] = [
+    fromStopPlace?.latitude || 0,
+    fromStopPlace?.longitude || 0,
+  ];
+
+  const mainStartIndex = findIndex(points, fromCoordinates);
+  const mainEndIndex = points.length - 1;
+
+  const beforeLegCoords = points.slice(0, mainStartIndex + 1);
+  const mainLegCoords = points.slice(mainStartIndex, mainEndIndex + 1);
+  const afterLegCoords = points.slice(mainEndIndex);
```

Comment:
FYI, the `afterLegCoords` is not really needed before we start working on the trip details page.